tests/modules/composite_reference_test/composite_reference_test.module

```diff
@@ -76,3 +77,20 @@ function composite_reference_test_entity_base_field_info(EntityTypeInterface $en
 
   return $fields;
 }
+
+/**
+ * Implements hook_query_TAG_alter().
+ */
+function composite_reference_test_query_entity_query_alter(SelectInterface $query) {
```

Comment:
Why are you complicating this so much? Why not instead implement node_query_node_access_alter() which is already directly a tag query alter for the node_access tag..and inside just do condition('nid', 0) or something simple (i guess the base table should already be available). Anything that kills the query. And then in the test don't log in as admin as it might skip the query access just because the user is admin..

The whole point is to assert that your query alter doesn't kick in for composite references.

Also, please have a commit showing the failure in drone with the test and one that fixes it so we can see that it actually works.

Thx